bot/cogs/error_handler.py

```diff
@@ -87,9 +88,29 @@ def __init__(self, bot: Bot):
                     return
 
                 # Return to not raise the exception
-                with contextlib.suppress(ResponseCodeError):
-                    await ctx.invoke(tags_get_command, tag_name=ctx.invoked_with)
+                tags_cog = self.bot.get_cog("Tags")
+                sent = await tags_cog._get_command(ctx, command_name)
+                # sent = await tags_get_command.callback(tags_get_command.cog, ctx, ctx.invoked_with)
+                if sent:
                     return
+                # No similar tag found, or tag on cooldown -
+                # searching for a similar command
+                raw_commands = [
+                    (cmd.name, *cmd.aliases)
+                    for cmd in self.bot.walk_commands()
+                    if not cmd.hidden
+                ]
+                raw_commands = [c for data in raw_commands for c in data]
+                similar_command_data = difflib.get_close_matches(command_name, raw_commands, 1)
+                similar_command_name = similar_command_data[0]
+                similar_command = self.bot.get_command(similar_command_name)
+                if similar_command.can_run(ctx):
+                    misspelled_content = ctx.message.content
+                    await ctx.send(
+                        f"Did you mean:\n**{misspelled_content.replace(command_name, similar_command_name)}**",
```

Comment:
This could mistakenly replace something other than the command name too. Should use a more precise method of replacing which only affects the command name.